play.py

```diff
@@ -63,25 +81,32 @@ def main(self):
                     self.run_AI_turn(curr_player)
                 self.turnNum += 1
 
-
+    # Defines logic for the first two turns where players select their settlements
     def firstTwoTurns(self):
         for i in range(4):
-            currTurn = self.turnNum % self.num_players
-            self.turnNum += 1
-            player = self.players[currTurn]
-            playerName = player.getName()
-            print("It is " + playerName + "\'s turn:")
-            
+            self.initial_settlement_placements(i)
+
+        for i in range(3, 0, -1):
+            self.initial_settlement_placements(i)
         return True
 
+    # This will place the initial settlement given the current player
+    def initial_settlement_placements(self, playerIndex):
+        player = self.players[playerIndex]
+        playerName = player.getName()
+        print("It is " + playerName + "\'s turn:")
+        possible_placement = self.game.getSettlementLocations(player)
```

Comment:
might have issue using get settlement locationsbecause that will eventually check if a road ends there. i can add a pregame bool to that function to handle it though

Reply:
Fair enough, I'll pass a boolean (firstTurn = true) and remove the road location

Reply:
Looks good to me.